src/react-components/home/usePaginatedAPI.js

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useState, useCallback, useRef } from "react";
+
+export function usePaginatedAPI(apiCallback) {
+  const curApiCallback = useRef(apiCallback);
+
+  const [state, setState] = useState({
+    isLoading: true,
+    hasMore: false,
+    results: [],
+    suggestions: [],
+    error: undefined,
+    cursor: 0,
+    nextCursor: undefined
+  });
+
+  const [internalState, setInternalState] = useState({ cursor: 0 });
+
+  const loadMore = useCallback(
+    () => {
+      if (state.nextCursor && !state.isLoading) {
```

Comment:
Might be worth warning here just so its not silently ignored. Though I suspect we will for the most part end up encapsulating usages of `loadMore` into paging buttons that also respect the loading state `eventually`, so it probably should never happen.